tests/test_dragon_installer.py

```diff
@@ -0,0 +1,519 @@
+            "is_crayex_platform",
+            lambda: is_crayex,
+        )
+        ctx.setattr(
+            smartsim._core.entrypoints.dragon_install,
+            "dragon_pin",
+            lambda: dragon_pin,
+        )
+        chosen_asset = filter_assets(asset_list)
+
+        if is_found:
+            assert chosen_asset
+            assert pyv in chosen_asset.name
+            assert dragon_pin in chosen_asset.name
+
+            if is_crayex:
+                assert "crayex" in chosen_asset.name.lower()
+        else:
+            assert not chosen_asset
+
+
+@pytest.mark.parametrize(
+    "dragon_pin,pyv,is_found,is_crayex",
+    [
+        pytest.param("dragon-0.8", "py3.8", False, False, id="0.8,python 3.8"),
+        pytest.param("dragon-0.8", "py3.9", True, False, id="0.8,python 3.9"),
+        pytest.param("dragon-0.8", "py3.10", True, False, id="0.8,python 3.10"),
+        pytest.param("dragon-0.8", "py3.11", True, False, id="0.8,python 3.11"),
+        pytest.param("dragon-0.8", "py3.12", False, False, id="0.8,python 3.12"),
+        pytest.param("dragon-0.8", "py3.8", False, True, id="0.8,python 3.8,CrayEX"),
+        pytest.param("dragon-0.8", "py3.9", True, True, id="0.8,python 3.9,CrayEX"),
+        pytest.param("dragon-0.8", "py3.10", True, True, id="0.8,python 3.10,CrayEX"),
+        pytest.param("dragon-0.8", "py3.11", True, True, id="0.8,python 3.11,CrayEX"),
+        pytest.param("dragon-0.8", "py3.12", False, True, id="0.8,python 3.12,CrayEX"),
+        pytest.param("dragon-0.9", "py3.8", False, False, id="0.9,python 3.8"),
+        pytest.param("dragon-0.9", "py3.9", True, False, id="0.9,python 3.9"),
+        pytest.param("dragon-0.9", "py3.10", True, False, id="0.9,python 3.10"),
+        pytest.param("dragon-0.9", "py3.11", True, False, id="0.9,python 3.11"),
+        pytest.param("dragon-0.9", "py3.12", False, False, id="0.9,python 3.12"),
+        pytest.param("dragon-0.9", "py3.8", False, True, id="0.9,python 3.8,CrayEX"),
+        pytest.param("dragon-0.9", "py3.9", True, True, id="0.9,python 3.9,CrayEX"),
+        pytest.param("dragon-0.9", "py3.10", True, True, id="0.9,python 3.10,CrayEX"),
+        pytest.param("dragon-0.9", "py3.11", True, True, id="0.9,python 3.11,CrayEX"),
+        pytest.param("dragon-0.9", "py3.12", False, True, id="0.9,python 3.12,CrayEX"),
+        # add a couple variants for a dragon version that isn't in the asset list
+        pytest.param("dragon-0.7", "py3.9", False, False, id="0.7,python 3.9"),
+        pytest.param("dragon-0.7", "py3.9", False, True, id="0.7,python 3.9,CrayEX"),
+    ],
+)
+def test_retrieve_asset_info(
+    asset_list: t.Dict[str, GitReleaseAsset],
+    monkeypatch: pytest.MonkeyPatch,
+    dragon_pin: str,
+    pyv: str,
+    is_found: bool,
+    is_crayex: bool,
+) -> None:
+    """Verify that an asset list is filtered correctly based on the python
+    version, platform (e.g. CrayEX, !CrayEx), and target dragon pin"""
```

Comment:
Doc string looks identical to the previous test

Reply:
removing the lower-level one since the second is a superset